breakerbp/failure_ratio.go

```diff
@@ -0,0 +1,131 @@
+package breakerbp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/sony/gobreaker"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+// FailureRatioBreaker is a circuit breaker based on gobreaker that uses a low-water-mark and
+// % failure threshold to trip.
+type FailureRatioBreaker struct {
+	goBreaker *gobreaker.CircuitBreaker
+
+	name              string
+	minRequestsToTrip int
+	failureThreshold  float64
+	logger            log.Wrapper
+}
+
+// Config represents the configuration for a FailureRatioBreaker.
+type Config struct {
+	// Minimum requests that need to be sent during a time period before the breaker is eligible to transition from closed to open.
+	MinRequestsToTrip int
+
+	// Percentage of requests that need to fail during a time period for the breaker to transition from closed to open.
+	// Represented as a float in [0,1], where .05 means >=5% failures will trip the breaker.
```

Comment:
🔕  Style nit: I _thiiiiink_ there should be a blank line between the first "sentence" in a docstring and the more detailed parts that come after?

Reply:
it feels weird to me to have the extra space in all of these so I guess I would prefer to leave it as-is? IDK I will admit I end up writing these in javadoc style often.

Reply:
I think it's _probably_ fine either way?  I probably inherited the extra blank line from Python docstrings so the lack of one looks weird to _me_ 😅

Reply:
yea that's not required by go's doc string, and I don't think the first paragraph summary is used anywhere (unlike git's commit message).